### ISSUE TITLE
Profiles not being listed

### ISSUE THREAD

**Issue by dannymidnight** [99designs/aws-vault#14]
Running `aws-vault ls` is always outputting "No credentials found". I assume this should be listing profiles that I've added into the vault.

**Comment by bradfeehan** [comment]
Works for me! ¯\\\_(ツ)_/¯

**Comment by lox** [comment]
Mind running it with `aws-vault --debug ls` and posting here? Also what version are you on?

**Comment by dannymidnight** [comment]
I'm on `v2.1.2` and running with the debug flag still only outputs `No credentials found`.

**Comment by karlbright** [comment]
I am having the same issue. I used the experimental posted today to create add some credentials. Modified my `~/.aws/config` and tested both accounts. They work fine but have run in to the same issue trying to list them both.

One thing to note is that I don't believe my ~/.aws/config existed before using the `aws-vault` command to add the first credentials and therefore create my keychain. I can't imagine this woudl be an issue but it is the only thing that I can think is not part of the typical process for most users.

**Comment by lox** [comment]
Hmmm, interesting. The ls command just lists entries in your keychain. I'd be interested to see what's actually in there for you guys. This is what mine looks like:

![keychain access](http://lachlan.me/s/t34Z0.png)

**Comment by karlbright** [comment]
I hadn't even thought to check that. Very interesting, I don't seem to have an aws-vault keychain at all.

**Comment by lox** [comment]
You'll need to right click under keychains and add it.

**Comment by karlbright** [comment]
That fixes the issue with `ls`. :+1:

**Comment by lox** [comment]
Wow, really?? That's very odd.

**Comment by lox** [comment]
Hrm, on second thought, perhaps it does make sense. Maybe the search I am using relies on a keychain being added to Keychain Access.

**Comment by karlbright** [comment]
@lox Yeh, would this have anything to do with only clicking "allow" when prompted by Keychain to allow access? I'm not entirely sure how Keychain Access works in OSX, so I could be way off here. I never clicked "always allow" when using aws-vault.

**Comment by karlbright** [comment]
On second thought, that shouldn't affect it at all? You probably have a better idea than what I do in this area. Glad theres a fix for it. I guess a short term solution is adding some README or prompt when a keychain is created?

**Comment by pda** [comment]
My old version of `aws-vault` (5a4d9c1fd37a59bf82ff142ccc2c09d94b239efb just after `v2` was merged) was successfully finding my one `99designs` credential; I've never added `aws-vault.keychain` to the Keychain Access app, nor clicked Always Allow.

The latest `master` and v2.2.0 and v2.3.0-experimental instead list all my `aws-keychain` credentials; I never imported them and I can't quite figure out where it's finding them. `dtruss` shows `open("/Users/pda/Library/Keychains/aws-keychain.keychain\0", 0, 0)` but it also opens all the other keychains.

After adding `aws-vault.keychain` to Keychain Access, `aws-vault ls` lists both my `aws-keychain` items plus the one `99designs` credential which I added to `aws-vault`. So `99designs` is listed twice.

I guess that's another confirmation that adding the keychain file to Keychain Access makes a difference.

¯\\\_(ツ)\_/¯
